src/functions.jl

```diff
@@ -679,6 +679,26 @@ struct VectorQuadraticFunction{T} <: AbstractVectorFunction
     constants::Vector{T}
 end
 
+function VectorQuadraticFunction(
+    rows::AbstractVector{ScalarQuadraticFunction{T}},
+) where {T}
+    ret = VectorQuadraticFunction{T}(
+        VectorQuadraticTerm{T}[],
+        VectorAffineTerm{T}[],
+        T[],
+    )
```

Comment:
I mentioned in the other PR also:
If this constructor is just for testing purposes and manually building constraints, I would be okay with the simple loop.
But, `vectorize` is used in bridges in potentially large problems.
Hence, we need, at least, some `sizehint!`s arounds